idea/src/org/jetbrains/kotlin/idea/inspections/collections/SimplifiableCallChainInspection.kt

```diff
@@ -36,6 +42,15 @@ class SimplifiableCallChainInspection : AbstractCallChainChecker() {
                         }
                     ) return@check false
                 }
+                if (conversion.replacement == "maxBy" || conversion.replacement == "minBy") {
+                    val argument = firstResolvedCall.valueArguments.firstOrNull()?.value?.arguments?.firstOrNull()
+                    val returnType = when (val argumentExpression = argument?.getArgumentExpression()) {
+                        is KtLambdaExpression -> argumentExpression.bodyExpression?.statements?.lastOrNull()?.getType(context)
```

Comment:
I don't like this `statements?.lastOrNull()`. This code isn't supposed to work for such cases:
```
fun main() {
    listOf(1, null, 2).sortedBy {
        if (true) return@sortedBy it
        it!!
    }.first()
}
```

Even though test works in general because lambdas with return statement [are discarded](https://github.com/JetBrains/kotlin/blob/76ac4ffb96c45953ea11c51f9aa7f249a94cfee8/idea/src/org/jetbrains/kotlin/idea/inspections/collections/AbstractCallChainChecker.kt#L46)

But you relay on behavior which is placed somewhere else inside `qualifiedExpressionVisitor` function => code locality is broken.

Please, use not latest statement `statements?.lastOrNull()` but actual return type of lambda

Reply:
`argumentExpression.bodyExpression?.statements?.lastOrNull()?.getType(context)` always returns nullable type.
How do we get actual return type of lambda without using last statement?

Reply:
@t-kameyama Why not just check the resulting descriptor parameter type?
`firstResolvedCall.resultingDescriptor.valueParameters`

Reply:
@yanex Fixed in 5e8d69e, PTAL.